//gRPC中推荐使用proto3语法
syntax = "proto3";

//enum可以定义一个选项菜单
enum Corpus {
    UNIVERSAL = 0;
    WEB = 1;
    IMAGES = 2;
    LOCAL = 3;
    NEWS = 4;
    PRODUCTS = 5;
    VIDEO = 6;
}

//使用message定义数据类型，相当于python中的类
message UserRequest {
    string user_id=1;
    int32 channel_id=2;
    int32 article_num=3;
    int64 time_stamp=4;
//    Corpus corpus = 5;
}


message Track {
    string click=1;
    string collect=2;
    string share=3;
    string read=4;
}

message Article {
    int64 article_id=1;
    Track track=2;
}

message ArticleResponse {
    string exposure=1;
    int64 time_stamp=2;
    repeated Article recommends=3;
}


//声明rpc调用的服务
service UserRcommend {
    rpc user_recommend(UserRequest) returns (ArticleResponse) {};
//    rpc article() returns () {};
}